lib.coffee

```diff
@@ -9,11 +9,30 @@ GZIP_HEADER = new Buffer([ 0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
 # DEFLATE ending block
 DEFLATE_END = new Buffer([ 0x03, 0x00 ])
 
+# Use the logic briefly described here by the author of zlib library:
+# http://stackoverflow.com/questions/14744692/concatenate-multiple-zlib-compressed-data-streams-into-a-single-stream-efficient#comment51865187_14744792
+# to generate deflate streams that can be concatenated into a gzip stream
 exports.createDeflatePart = ->
+	buf = new Buffer(0) # buffer each chunk until we are sure it is not the last one
 	compress = new DeflateCRC32Stream()
+	compress._push = compress.push
```

Comment:
For backwards compatibility it would be: `exports.createDeflatePart = -> new DeflatePartStream()`, although you could also expose the class directly and deprecate `createDeflatePart` for removal in a major version